### PR TITLE
refactor: organization profile APIs

### PR DESCRIPTION
**Pull request by pranalidhanavade**
Refactored following APIs from organization module:

- GET `/orgs/public-profiles/{orgSlug}` Fetch organization details.
- GET `/orgs/dashboard/{orgId}` Get dashboard details.
- GET `/orgs/{orgId}` Get an organization.

### What ###
- Ensured consistency of API status codes.
- Handled error and success messages.
- Tested with both positive and negative scenarios.

### How ###
- Checked and standardized API status codes to ensure a uniform response.

### Why ###
- Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### CODE REVIEW

File: apps/organization/interfaces/organization.interface.ts [hunk 1]
```diff
@@ -119,13 +119,6 @@ interface IOrganizationPagination {
   logoUrl: string;
 }
 
-export interface IOrganizationDashboard {
-  usersCount: number,
-  schemasCount: number,
```
Comment (nishad-ayanworks) [inline apps/organization/interfaces/organization.interface.ts:124]: why did you remove this. Could you please tell me?

**Review by nishad-ayanworks — APPROVED** [review]
LG

File: apps/organization/interfaces/organization.interface.ts (hunk 1 above)
Comment (pranalidhanavade) [reply apps/organization/interfaces/organization.interface.ts:124]: We are using this interface in api-gateway and organisation service, so I moved it to a common file.

**Review by ankita-p17 — APPROVED** [review]
